fortuna/src/api.rs

```diff
@@ -137,3 +153,229 @@ impl IntoResponse for RestError {
         }
     }
 }
+
+pub fn routes(state: ApiState) -> Router<(), Body> {
+    Router::new()
+        .route("/", get(index))
+        .route("/live", get(live))
+        .route("/metrics", get(metrics))
+        .route("/ready", get(ready))
+        .route("/v1/chains", get(chain_ids))
+        .route(
+            "/v1/chains/:chain_id/revelations/:sequence",
+            get(revelation),
+        )
+        .with_state(state)
+}
+
+#[cfg(test)]
+mod test {
+    use {
+        crate::{
+            api::{
+                self,
+                ApiState,
+                BinaryEncoding,
+                Blob,
+                BlockchainState,
+                GetRandomValueResponse,
+            },
+            chain::reader::mock::MockEntropyReader,
+            state::{
+                HashChainState,
+                PebbleHashChain,
+            },
+        },
+        axum::{
+            http::StatusCode,
+            response,
+        },
+        axum_test::{
+            TestResponse,
+            TestServer,
+        },
+        ethers::prelude::Address,
+        once_cell::sync::Lazy,
+        std::sync::Arc,
+    };
+
+    const PROVIDER: Address = Address::zero();
+    const OTHER_PROVIDER: Lazy<Address> = Lazy::new(|| Address::from_low_u64_be(1));
+    const ETH_CHAIN: Lazy<Arc<HashChainState>> = Lazy::new(|| {
```

Comment:
The problem with having these as Arc's is their state will persist between other tests. In general I think It's a bad idea to do this, especially as Rust runs tests asynchronously leading to non-deterministic and unpredictable behaviour. I think you're better off moving these inside your test_server() initializer and returning them there, or just setting them up at the start of the test.